src/main/java/io/xdb/core/process/Process.java

```diff
@@ -0,0 +1,18 @@
+package io.xdb.core.process;
+
+import io.xdb.core.state.StateSchema;
+import io.xdb.core.utils.ClassUtil;
+
+/**
+ * The {@link Process} interface is used to define a process definition.
+ * It represents a fundamental concept at the top level in XDB.
+ */
+public interface Process {
+    default String getType() {
```

Comment:
I am thinking to do this for Golang SDK recently -- 

What do you think that we move the `ProcessOptions` here so that ProcessOption will include this Type, and others like timeouts, IdReusePolicy? 

Because usually those timeout/IdReusePolicy can be determined when implementing the process definition, rather than being specified when starting a process. 

Another reasons is that I am hoping to have a more extensible struct to to represent those "optional configuration" for a process definition.

Reply:
That's a good idea. I am going to implment in this way here.

Reply:
Just tried in Java and found that if moving the type out of Process, we cannot provide the default value as the `class.getSimpleName()` because now the type is not included in the class def.

Reply:
We can do it in the Golang way — use an empty value to represent the default value. When resolving the process type , we can use class name

Reply:
Makes sense.